bot/exts/utilities/epoch.py

```diff
@@ -0,0 +1,123 @@
+from typing import Union
+
+import arrow
+import dateutil
+import discord
+from dateutil import parser
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.exts.core.extensions import invoke_help_command
+
+# https://discord.com/developers/docs/reference#message-formatting-timestamp-styles
+TIMESTAMP_FORMATS = [
+    "h:mm A",
+    "h:mm:ss A",
+    "MM/DD/YYYY",
+    "MMMM D, YYYY",
+    "MMMM D, YYYY h:mm A",
+    "dddd, MMMM D, YYYY h:mm A "
+]
+STYLES = {"Epoch": "",
+          "Short Time": "t",
+          "Long Time": "T",
+          "Short Date": "d",
+          "Long Date": "D",
+          "Short Date/Time": "f",
+          "Long Date/Time": "F",
+          "Relative Time": "R"
+          }
+
+
+class RelativeDate(commands.Converter):
+    async def convert(self, ctx: commands.Context, argument: str) -> arrow.Arrow:
+        return arrow.utcnow().dehumanize(argument)
+
+
+class AbsoluteDate(commands.Converter):
+    async def convert(self, ctx: commands.Context, argument: str) -> arrow.Arrow:
+        return arrow.get(dateutil.parser.parse(argument))
+
+
+class Epoch(commands.Cog):
+
+    @commands.command(name="epoch")
+    async def epoch(self, ctx: commands.Context, *, date_time: Union[RelativeDate, AbsoluteDate] = None) -> None:
+        """
+        Convert an entered time and date to a unix timestamp. Both relative and absolute times are accepted.
+
+        **Relative time**
+            accepted units: "seconds", "minutes", "hours", "days", "weeks", "months", "years"
+            Eg ".epoch in a month 4 days and 2 hours"
+
+        **Absolute time**
+            eg ".epoch 2022/6/15 16:43 -04:00"
+            Absolute times must be entered in descending orders of magnitude.
+            Timezones may take the following forms:
+                Z (UTC)
```

Comment:
It is a bit sad that you have to explicitly mention `Z`, would there be any way around this one?

Reply:
Writing the timezone is optional, it will default to UTC anyway. `Z` and `UTC` should both be recognized as well, but other named timezones (PST, EST, etc.) won't be. Do you think I should remove that line from the docstring?

Reply:
Oh I see, yeah you could mention in the help text that the timezone is optional and will default to UTC